### PR TITLE
feat: options for miner and optimizer

### PR DESCRIPTION
**Pull request by guenthermi**
Adds support for miner and optimizer options and documentation about optimizers and miners.

Addresses [#299](https://github.com/jina-ai/finetuner-core/issues/299) and [#300](https://github.com/jina-ai/finetuner-core/issues/300)

---

- [x] This PR references an open issue
- [x] I have added a line about this change to CHANGELOG

### CODE REVIEW

File: docs/walkthrough/run-job.md [hunk 1]
```diff
@@ -59,6 +62,7 @@ run = finetuner.fit(
     model_options={}, # additional options to pass to the model constructor
     loss='TripletMarginLoss', # Use CLIPLoss for CLIP fine-tuning.
     miner='TripletMarginMiner',
+    miner_options={'margin': 0.2}, # additional options for the miner constructor
```
Comment (bwanglzu) [inline docs/walkthrough/run-job.md:65]: can you add optimizer options to here?

File: docs/walkthrough/run-job.md [hunk 2]
```diff
@@ -73,4 +77,31 @@ run = finetuner.fit(
...
+If you want to pass more parameters to the optimizer, you can specify them via `optimizer_options`.
+For example, you can enable the weight decay of the Adam optimizer to penalize high weights in the model by setting `optimizer_options={'weight_decay':0.01}`.
+
+For detailed documentation of the optimizers and their parameters, please take a look at the [PyTorch documentation](https://pytorch.org/docs/stable/optim.html).
+
+### Configuration of the Miner
+
+To filter the instances in a batch that are used to calculate the loss, you can use miners.
+Finetuner allows you to use miners provided by the [Pytorch Metric Learning](https://kevinmusgrave.github.io/pytorch-metric-learning) framework. 
+To select a specific miner, you can pass its name to the fit function, e.g., `AngularMiner`, `TripletMarginMiner`, ...
```
Comment (bwanglzu) [inline docs/walkthrough/run-job.md:98]: you can list all miners to the developer reference field, all documented in docstring

**Review by bwanglzu — APPROVED** [review]
LGTM!

File: docs/walkthrough/run-job.md (hunk 2 above)
Comment (guenthermi) [reply docs/walkthrough/run-job.md:98]: First I had this, but it looked ugly, because the names are so long and therefore, the list is very long.

**Review by gmastrapas — APPROVED** [review]
Looks good!

**Pull request merged**